complex.ts

```diff
@@ -0,0 +1,43 @@
+export class Complex {
+  real: number;
+  imag: number;
+  constructor(cmplx: Complex);
+  constructor(real?: number, imag?: number);
+  constructor(realOrCmplx: Complex | number = 0, imag = 0) {
+    if (realOrCmplx instanceof Complex) {
+      this.real = realOrCmplx.real;
+      this.imag = realOrCmplx.imag;
+    } else {
+      this.real = realOrCmplx;
+      this.imag = imag;
+    }
+  }
+
+  add(v: number | Complex) {
+    if (v instanceof Complex) {
+      this.real += v.real;
+      this.imag += v.imag;
+    } else {
+      this.real += v;
```

Comment:
BTW. Does this have a prototype reference?

I haven't seen anything special about this writing.

And `this.real += v;` This is not a high-precision calculation

Reply:
see [julia](https://github.com/JuliaLang/julia/blob/b75fce29860285102c3929f2e101aaea00a4077d/base/complex.jl)

Reply:
Complex numbers have some different rules than real numbers